src/client.rs

```diff
@@ -160,38 +251,272 @@ where
 
-                // FIXME: should we also the root as `MetadataVersion::None`?
+        // Only store the metadata after we have validated it.
+        if fetched {
+            let root_version = MetadataVersion::Number(root.version());
+            local
+                .store_metadata(&root_path, &root_version, &root)
+                .await?;
 
-                root
-            }
-        };
+            // Also store this root metadata as the latest version.
+            local
+                .store_metadata(&root_path, &MetadataVersion::None, &root)
+                .await?;
+        }
+
+        Ok(Client {
+            tuf,
+            config,
+            local,
+            remote,
+        })
+    }
 
-        let tuf = Tuf::from_root_pinned(root, trusted_root_keys)?;
+    /// Create a new TUF client. It will attempt to load initial root metadata from the local and remote
+    /// repositories using the provided keys to pin the verification.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use chrono::offset::{Utc, TimeZone};
+    /// # use futures_executor::block_on;
+    /// # use std::iter::once;
+    /// # use tuf::{
+    /// #     Error,
+    /// #     interchange::Json,
+    /// #     client::{Client, Config},
+    /// #     crypto::{KeyType, PrivateKey, SignatureScheme},
+    /// #     metadata::{MetadataPath, MetadataVersion, Role, RootMetadataBuilder},
+    /// #     repository::{EphemeralRepository, Repository},
+    /// # };
+    /// # fn main() -> Result<(), Error> {
+    /// # block_on(async {
+    /// # let private_key = PrivateKey::from_pkcs8(
+    /// #     &PrivateKey::new(KeyType::Ed25519)?,
+    /// #     SignatureScheme::Ed25519,
+    /// # )?;
+    /// # let public_key = private_key.public().clone();
+    /// let local = EphemeralRepository::new();
+    /// let remote = EphemeralRepository::new();
+    ///
+    /// let root_version = 1;
+    /// let root_threshold = 1;
+    /// let root = RootMetadataBuilder::new()
+    ///     .version(root_version)
+    ///     .expires(Utc.ymd(2038, 1, 1).and_hms(0, 0, 0))
+    ///     .root_key(public_key.clone())
+    ///     .root_threshold(root_threshold)
+    ///     .snapshot_key(public_key.clone())
+    ///     .targets_key(public_key.clone())
+    ///     .timestamp_key(public_key.clone())
+    ///     .signed::<Json>(&private_key)?;
+    ///
+    /// let root_path = MetadataPath::from_role(&Role::Root);
+    /// let root_version = MetadataVersion::Number(root_version);
+    ///
+    /// remote.store_metadata(&root_path, &root_version, &root).await?;
+    ///
+    /// let client = Client::from_pinned_root_keys(
+    ///     Config::default(),
+    ///     &root_version,
+    ///     root_threshold,
+    ///     once(&public_key),
+    ///     local,
+    ///     remote,
+    /// ).await?;
+    /// # Ok(())
+    /// # })
+    /// # }
+    /// ```
+    pub async fn from_pinned_root_keys<'a, I>(
```

Comment:
I think "from_pinned_root_keys" isn't a very descriptive name for this constructor, since it doesn't really come "from" the root keys. It's really `from_local_or_remote_with_pinned_root_keys`, but that name is probably too long. `with_pinned_root_keys` seems like an acceptable shorthand to me, but my feelings aren't that strong on the matter. Feel free to get a second opinion.

Reply:
`with_pinned_root_keys` seems fine to me. I also updated `Tuf` to reflect these names.